products/accessapproval/api.yaml

```diff
@@ -0,0 +1,240 @@
+                  all
+                  appengine.googleapis.com
+                  bigquery.googleapis.com
+                  bigtable.googleapis.com
+                  cloudkms.googleapis.com
+                  compute.googleapis.com
+                  dataflow.googleapis.com
+                  iam.googleapis.com
+                  pubsub.googleapis.com
+                  storage.googleapis.com
+            - !ruby/object:Api::Type::Enum
+              name: enrollmentLevel
+              description: |
+                The enrollment level of the service.
+              default_value: :BLOCK_ALL
+              values:
+                - :BLOCK_ALL
+      - !ruby/object:Api::Type::Boolean
+        name: enrolledAncestor
+        output: true
+        description: |
+          If the field is true, that indicates that at least one service is enrolled for Access Approval in one or more ancestors of the Folder.
+  - !ruby/object:Api::Resource
+    name: ProjectSettings
+    base_url: "projects/{{project}}/accessApprovalSettings"
+    self_link: 'projects/{{project}}/accessApprovalSettings'
+    create_verb: :PATCH
+    update_verb: :PATCH
+    update_mask: true
+    description: |
+      Access Approval enables you to require your explicit approval whenever Google support and engineering need to access your customer content.
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      api: 'https://cloud.google.com/access-approval/docs/reference/rest/v1/projects'
+    parameters:
+      - !ruby/object:Api::Type::String
+        name: project_id
+        description: |
+          ID of the project of the access approval settings.
+        required: true
+        input: true
+        url_param_only: true
+    properties:
+      - !ruby/object:Api::Type::String
+        name: name
+        description: |
+          The resource name of the settings. Format is "projects/{project_id/accessApprovalSettings"
+        output: true
+      - !ruby/object:Api::Type::Array
+        name: notificationEmails
+        description: |
+          A list of email addresses to which notifications relating to approval requests should be sent.
+          Notifications relating to a resource will be sent to all emails in the settings of ancestor
+          resources of that resource. A maximum of 50 email addresses are allowed.
+        max_size: 50
+        item_type: Api::Type::String
+      - !ruby/object:Api::Type::Array
+        name: enrolledServices
+        description: |
+          A list of Google Cloud Services for which the given resource has Access Approval enrolled.
+          Access requests for the resource given by name against any of these services contained here will be required
+          to have explicit approval. Enrollment can only be done on an all or nothing basis.
+          
+          If a cloudProduct is repeated in this list, the first entry will be honored and all following entries will be discarded.
```

Comment:
Does this statement apply to TF? Assuming we make this a set, what happens TF-wise when we define multiple identical entries in a config?